apps/coa/behaviours/save-document.js

```diff
@@ -3,11 +3,15 @@
 const config = require('../../../config');
 const Model = require('../models/file-upload');
 
+// Use regex to replace the middle part of the filename with **REDACTED**,
+// keeping the first 2 and last 2 characters before the extension
+const sanitiseFilename = filename => filename.replace(/^(.{2}).*(.{2}\.[^.]+)$/, '$1**REDACTED**$2');
```

Comment:
Nice

Reply:
If you like you could take this func out into a utility location as it's used two or three times across this code. 
Add some tests, etc.